✨ Feature: 트렌드 미션 내용 수정 API

Ko-udon opened the pull request:
## 개요
- 기능
  - 사용자는 트렌드 미션 내용 수정 가능
  - 본인의 트렌드 미션이 아닐 시, 요청 실패

- 비고
  - 테스트를 위해 multipart/form으로 데이터를 변환해서 전송 시도 -> 실패
  - 테스트에 너무 많은 시간이 소모되어 우선 직접 테스트한 캡처 사진으로 대체

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가


## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.  
- [x] 변경 사항에 대한 테스트를 사진으로 대체하겠습니다.

![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/9a918820-e40a-41d4-8624-103bf01465f1)
![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/3b0a854e-adc9-435f-a452-e10db0580e25)
![image](https://github.com/HotCake-Tanghuru/HotCake-BE/assets/79897135/51329afe-6a2a-4a36-b2c1-89e310981d83)


 Resolves: #20 #25

Ko-udon committed:
✨ Feature: 트렌드 미션 내용 수정 API

- 기능
  - 사용자는 트렌드 미션 내용 수정 가능
  - 본인의 트렌드 미션이 아닐 시, 요청 실패

- 비고
  - 테스트를 위해 multipart/form으로 데이터를 변환해서 전송 시도 -> 실패
  - 테스트에 너무 많은 시간이 소모되어 우선 직접 테스트한 캡처 사진으로 대체

nayeongdev commented on trend_missions/tests.py:
```diff
@@ -162,4 +167,66 @@ def test_TrendMissions_detail(self):
+        # 테스트를 위한 트렌드 아이템 생성
+        self.image = SimpleUploadedFile(
+            name='test_image.jpg', 
+            content=b'',
```
원래  `content=b'',` 이렇게 쓰는건가요..?

Ko-udon replied:
빈 바이트 파일을 나타낼때 사용한다고 하더군요.
테스트를 위해 가상의 이미지 파일을 할당하려고 했는데 테스트는 실패했습니다 ㅠㅠ..
테스트는 부득이하게 이미지로 대체하였습니다.

nayeongdev commented on trend_missions/tests.py:
```diff
@@ -162,4 +167,66 @@ def test_TrendMissions_detail(self):
+    #     data = request.body
+    #     content_type = request.content_type
+
+    #     response = self.client.put(f'/trend-missions/mission-item/{self.trend_mission_item.id}/edit', data, content_type=content_type)
```
`factory.put`과 `client.put`은 중복되는 과정인 것 같은데 따로따로 테스트 하신건가요??

Ko-udon replied:
테스트 코드 작성 도중, 이미지 캡처로 대처하다보니 코드에 신경을 못썼습니다. 테스트 코드를 아예 삭제하는게 나을 것 같습니다. 삭제해서 다시 커밋 하겠습니다.

Ko-udon committed:
♻️ Refactor: 요청처리 Patch로 변경

- 특정 리소스를 완전 대체하는 것이 아닌, 일부값을 변경하므로 Patch가 맞다고 판단하여 수정.

Ko-udon committed:
♻️ Refactor: 테스트 코드 삭제

- 테스트 이미지 캡처본으로 대체

Ko-udon committed:
♻️ Refactor: Trend-mission 모델의 FK 필드명을 _id 삭제

- Trend-mission 모델의 TrendMission, Comment, UserTrendItem, Stamp 필드명 모두 수정
- 필드명_id -> 필드명 형태로 수정

- 기존의 Test코드 수정 및 테스트 성공

Ko-udon committed:
Merge branch 'develop' into feat/trend-mission-edit
